tests/test_payments.py

```diff
@@ -273,3 +283,62 @@ def test_index(self):
         resp = self.app.get('/')
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
         self.assertTrue('Welcome to payments!' in resp.data)
+#test cases for update payments - put and patch
+# passing correct data to put
+    @mock.patch.object(PaymentService, 'update_payment', return_value=PUT_CREDIT_RETURN, autospec=True)
+    def test_crud_update_put(self,mock_ps_update):
+        data = json.dumps(PUT_CREDIT_RETURN)
+        resp = self.app.put('/payments/1', data=data, content_type='application/json')
+        mock_ps_update.assert_called_with(mock.ANY,1,payment_replacement=PUT_CREDIT_RETURN)
+        self.assertEqual( resp.status_code, status.HTTP_200_OK )
+        new_json = json.loads(resp.data)
+        self.assertEqual (new_json['nickname'], 'favcredit')
+
+# passing correct data to patch
+    @mock.patch.object(PaymentService, 'update_payment', return_value=PATCH_RETURN, autospec=True)
+    def test_crud_update_patch(self,mock_ps_update):
+        data = json.dumps(PATCH_CREDIT)
+        resp = self.app.patch('/payments/1', data=data, content_type='application/json')
+        mock_ps_update.assert_called_with(mock.ANY,1,payment_attributes=PATCH_CREDIT)
+        self.assertEqual( resp.status_code, status.HTTP_200_OK )
+        new_json = json.loads(resp.data)
+        self.assertEqual (new_json['nickname'], 'boringcredit')
+
+# passing text data to put
+    def test_crud_update_put_with_text_data(self):
+        resp = self.app.put('/payments/1', data="hello", content_type='text/plain')
+        self.assertEqual( resp.status_code, status.HTTP_400_BAD_REQUEST )
+        self.assertTrue('bad or no data' in resp.data)
```

Comment:
We should probably replace `bad or no data` with an error message constant defined in`payments.py`.  Just import the constant from `payments.py` and use it here.